server/channels/app/webhook_test.go

```diff
@@ -296,21 +296,25 @@ func TestCreateWebhookPost(t *testing.T) {
 	require.Nil(t, err)
 	defer th.App.DeleteIncomingWebhook(hook.Id)
 
-	post, err := th.App.CreateWebhookPost(th.Context, hook.UserId, th.BasicChannel, "foo", "user", "http://iconurl", "", model.StringInterface{
-		"attachments": []*model.SlackAttachment{
-			{
-				Text: "text",
+	post, err := th.App.CreateWebhookPost(th.Context, hook.UserId, th.BasicChannel, "foo", "user", "http://iconurl", "",
+		model.StringInterface{
+			"attachments": []*model.SlackAttachment{
+				{
+					Text: "text @all",
+				},
 			},
+			"webhook_display_name": hook.DisplayName,
 		},
-		"webhook_display_name": hook.DisplayName,
-	}, model.PostTypeSlackAttachment, "")
+		model.PostTypeSlackAttachment,
+		"", &model.PostPriority{RequestedAck: model.NewBool(true), Priority: model.NewString("high")})
 	require.Nil(t, err)
 
 	assert.Contains(t, post.GetProps(), "from_webhook", "missing from_webhook prop")
 	assert.Contains(t, post.GetProps(), "attachments", "missing attachments prop")
 	assert.Contains(t, post.GetProps(), "webhook_display_name", "missing webhook_display_name prop")
+	assert.True(t, *post.GetPriority().RequestedAck, "RequestedAck is not true")
```

Comment:
I'm not familiar with this feature, but in the webhook you are setting the `PostMetadata.Priority` and in the test you are checking the `PostMetadata.RequestedAck` field. 
Does setting post priority to high automatically set `RequestedAck`? 
Shouldn't we check that `Priority` is the one we set up? 
Should other paths be tested? (Where `Priority` is not `high`)

Reply:
> Does setting post priority to high automatically set RequestedAck?

confusingly, `postMetadata.priority` contains a nested struct with `priority`, `requested_ack` , and `persistant_notifications` - https://github.com/mattermost/mattermost/blob/9d6d8cceec5dfc1ec413741a9949be8cfa85c57a/server/public/model/post_metadata.go#L27

So, I'm setting the `postMetadata.priority` to equal the values they send in, which should match the struct above. Creating the weird nest.

> Shouldn't we check that Priority is the one we set up?

You're right, I added a test for priority to be equal. 

> Should other paths be tested? (Where Priority is not high)

I'm happy to add paths to test `urgent` and `requested_ack` too. I'm unsure how verbose we want to be here. My thought was to setup a test just to confirm the value is flowing through correctly. Let me make a new set of tests that handles all the priority things!